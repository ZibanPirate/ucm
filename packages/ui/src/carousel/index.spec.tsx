import { fireEvent, render, screen } from "@testing-library/react";

import { Carousel } from ".";

describe(`Testing component '${Carousel.name}' :`, () => {
  it(`should render it with 2 images, and switch between them`, () => {
    const { container } = render(
      <Carousel
        data-testid="carousel"
        previousProps={{ "data-testid": "carousel-previous" }}
        nextProps={{ "data-testid": "carousel-next" }}
        images={["/assets/test-01.svg", "/assets/test-02.svg"]}
      />,
    );
    expect(container).toMatchSnapshot("First image");

    const carouselNext = screen.getByTestId("carousel-next");
    fireEvent.click(carouselNext);
    expect(container).toMatchSnapshot("Second image");

    fireEvent.click(carouselNext);
    expect(container).toMatchSnapshot("First image again");

    const carouselPrevious = screen.getByTestId("carousel-previous");
    fireEvent.click(carouselPrevious);
    expect(container).toMatchSnapshot("Second image again");
  });

  it(`should render it with 2 images and no optional props`, () => {
    const { container } = render(
      <Carousel images={["/assets/test-01.svg", "/assets/test-02.svg"]} />,
    );
    expect(container).toMatchSnapshot();
  });
});
