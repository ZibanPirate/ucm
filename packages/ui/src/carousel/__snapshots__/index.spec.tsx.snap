// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing component 'Carousel' : should render it with 2 images and no optional props 1`] = `
<div>
  <div>
    <img
      src="/assets/test-01.svg"
      style="object-fit: cover; width: 100%;"
    />
    <div
      style="display: flex; flex-direction: row; flex: 1; margin: initial; justify-content: space-around;"
    >
      <button
        class="ucm-ui-button"
        style="font-size: medium; margin: initial;"
      >
        &lt;
      </button>
      <button
        class="ucm-ui-button"
        style="font-size: medium; margin: initial;"
      >
        &gt;
      </button>
    </div>
  </div>
</div>
`;

exports[`Testing component 'Carousel' : should render it with 2 images, and switch between them: First image 1`] = `
<div>
  <div
    data-testid="carousel"
  >
    <img
      src="/assets/test-01.svg"
      style="object-fit: cover; width: 100%;"
    />
    <div
      style="display: flex; flex-direction: row; flex: 1; margin: initial; justify-content: space-around;"
    >
      <button
        class="ucm-ui-button"
        data-testid="carousel-previous"
        style="font-size: medium; margin: initial;"
      >
        &lt;
      </button>
      <button
        class="ucm-ui-button"
        data-testid="carousel-next"
        style="font-size: medium; margin: initial;"
      >
        &gt;
      </button>
    </div>
  </div>
</div>
`;

exports[`Testing component 'Carousel' : should render it with 2 images, and switch between them: First image again 1`] = `
<div>
  <div
    data-testid="carousel"
  >
    <img
      src="/assets/test-01.svg"
      style="object-fit: cover; width: 100%;"
    />
    <div
      style="display: flex; flex-direction: row; flex: 1; margin: initial; justify-content: space-around;"
    >
      <button
        class="ucm-ui-button"
        data-testid="carousel-previous"
        style="font-size: medium; margin: initial;"
      >
        &lt;
      </button>
      <button
        class="ucm-ui-button"
        data-testid="carousel-next"
        style="font-size: medium; margin: initial;"
      >
        &gt;
      </button>
    </div>
  </div>
</div>
`;

exports[`Testing component 'Carousel' : should render it with 2 images, and switch between them: Second image 1`] = `
<div>
  <div
    data-testid="carousel"
  >
    <img
      src="/assets/test-02.svg"
      style="object-fit: cover; width: 100%;"
    />
    <div
      style="display: flex; flex-direction: row; flex: 1; margin: initial; justify-content: space-around;"
    >
      <button
        class="ucm-ui-button"
        data-testid="carousel-previous"
        style="font-size: medium; margin: initial;"
      >
        &lt;
      </button>
      <button
        class="ucm-ui-button"
        data-testid="carousel-next"
        style="font-size: medium; margin: initial;"
      >
        &gt;
      </button>
    </div>
  </div>
</div>
`;

exports[`Testing component 'Carousel' : should render it with 2 images, and switch between them: Second image again 1`] = `
<div>
  <div
    data-testid="carousel"
  >
    <img
      style="object-fit: cover; width: 100%;"
    />
    <div
      style="display: flex; flex-direction: row; flex: 1; margin: initial; justify-content: space-around;"
    >
      <button
        class="ucm-ui-button"
        data-testid="carousel-previous"
        style="font-size: medium; margin: initial;"
      >
        &lt;
      </button>
      <button
        class="ucm-ui-button"
        data-testid="carousel-next"
        style="font-size: medium; margin: initial;"
      >
        &gt;
      </button>
    </div>
  </div>
</div>
`;
